### ISSUE TITLE
Questions about the forward operation of img_u_w in training mode

### ISSUE THREAD

**Issue by Hugo-cell111** [LiheYoung/UniMatch#94]
Hi! As mentioned in [issue68](https://github.com/LiheYoung/UniMatch/issues/68), the img_u_w is to provide clean and informative BN statistics to the model, therefore you set model.train(). I wonder how you came up with such a method and how you obtain such a conclusion that it will provide clean and informative BN statistics? In common setting we won't send the image in the training mode, but applying such a skill indeed brings a promising performance boosting. Will it lead to information leakage? Thanks!

**Comment by LiheYoung** [comment]
From us, there were many private observations that strongly augmented images can severely affect the BN statistics. But honestly, I cannot clearly remember how we came up with the solution of feeding the weakly augmented images to the network in the train mode. We started to use this practice around 2 years ago. But actually, in the FixMatch implementations, they also feed the weakly augmented images to the network in the train mode. Maybe this gave us some insight at that time.
